network/network.go

```diff
@@ -369,30 +394,48 @@ func (n *Network) connectToKnownNodes(nodeDID did.DID) error {
 		return err
 	}
 	for _, node := range otherNodes {
-		if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
-			// Found local node, do not discover.
-			continue
-		}
-	inner:
-		for _, service := range node.Service {
-			if service.Type == transport.NutsCommServiceType {
-				var nutsCommUrl transport.NutsCommURL
-				if err = service.UnmarshalServiceEndpoint(&nutsCommUrl); err != nil {
-					log.Logger().
-						WithError(err).
-						WithField(core.LogFieldDID, node.ID.String()).
-						Warn("Failed to extract NutsComm address from service")
-					continue inner
-				}
+		n.connectToDID(nodeDID, node)
+	}
+
+	// Assume this is a new node when no NutsComm endpoints are found.
+	n.assumeNewNode = len(otherNodes) == 0
+	if n.assumeNewNode {
+		log.Logger().Infof("assuming this is a new node, discovered NutsComm addresses are processed with a %s delay", newNodeConnectionDelay)
+	}
+
+	return nil
+}
+
+func (n *Network) connectToDID(nodeDID did.DID, node did.Document) {
+	if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
+		// Found local node, do not discover.
```

Comment:
add note for clustering?

Reply:
what would be in the note? 
The nodes in the cluster should still not connect to their own DID using the NutsComm address?